src/dist/http/pprof_http_service.cpp

```diff
@@ -320,25 +321,32 @@ void pprof_http_service::symbol_handler(const http_request &req, http_response &
 // == ip:port/pprof/heap == //
 //                          //
 
-static constexpr const char *TCMALLOC_SAMPLE_PARAMETER = "TCMALLOC_SAMPLE_PARAMETER";
-
-static bool is_heap_profile_enabled() { return ::getenv(TCMALLOC_SAMPLE_PARAMETER) != nullptr; }
-
-static bool get_heap_profile(std::string &result)
+void pprof_http_service::heap_handler(const http_request &req, http_response &resp)
 {
-    if (!is_heap_profile_enabled()) {
-        result = "no TCMALLOC_SAMPLE_PARAMETER in env";
-        return false;
+    const std::string SECOND = "seconds";
```

Comment:
We should prevent this function concurrent invoking.

Reply:
Yes, you are right~ We should prevent this function not executing concurrently, and we may should also prevent pprof function executing concurrently. For example, we should not try to get heap profile and get growth at the same time. I will add a TODO comment here, and research about it.